src/provider/store/Store.ts

```diff
@@ -2,52 +2,56 @@ import {Signal, Aggregators} from 'openfin-service-signal';
 
 import {AsyncInit} from '../controller/AsyncInit';
 
-/**
- * Subset of properties of `Store` that are safe for use from actions and components.
- */
-export interface StoreAPI<S, A> {
-    state: S;
-    dispatch(action: A): Promise<void>;
-}
-
 export abstract class Action<S> {
     public readonly type?: string;
 
     constructor(type?: string) {
```

Comment:
Either specify `type` for all actions (could aid with debugging), or remove this optional field and add only to the `Init` action.

Reply:
Can you change the type definition in the base class to make `type` abstract, rather than initialising to `this.constructor.name`?

Will ensure that all sub-classes specify `type`, which should keep the values consistent.